### PR TITLE
README Updates - Addresses #8

### PR DESCRIPTION
**Pull request by NatoliChris**
This PR will update the README in `ssz` according to Issue #8. This PR will add the following: 
* Building 
* Dependencies
* Overview
* Examples

(WIP)

### CODE REVIEW

**Comment by NatoliChris** [comment]
**Progress Update**

WIP status blocked; Waiting for `decode` to be completed so the example usage can be correctly detailed. :smile:

**Comment by NatoliChris** [comment]
No rush at all :smile: 
Thanks for your time @paulhauner :) I appreciate it!

File: ssz/README.md [hunk 1]
```diff
@@ -1,21 +1,549 @@
...
-serialization described by Vitalik Buterin. The method is tentatively intended
-for use in the Ethereum Beacon Chain.
+This is currently a ***Work In Progress*** crate.
+
+
+SimpleSerialize is a serialization protocol described by Vitalik Buterin. The
+method is tentatively intended for use in the Ethereum Beacon Chain as
+described in the [Ethereum 2.1 Spec](https://notes.ethereum.org/s/Syj3QZSxm).
 
 There are two primary sources for this spec, and they are presently
```
Comment (paulhauner) [inline ssz/README.md]: I think we should indicate that we are following the beacon chain spec and that we completely disregard the other spec. In other words, communicate that the beacon chain spec is canonical.

File: ssz/README.md [hunk 2]
```diff
@@ -1,21 +1,549 @@
...
+  + [decode_ssz(ssz_bytes: &(u8), index: usize) -> Result](#decode_sszssz_bytes-u8-index-usize---resultt-usize-decodeerror)
+  + [decode_ssz_list(ssz_bytes: &(u8), index: usize) -> Result, usize), DecodeError>](#decode_ssz_listssz_bytes-u8-index-usize---resultvec-usize-decodeerror)
+  + [decode_length(bytes: &(u8), index: usize, length_bytes: usize) -> Result](#decode_lengthbytes-u8-index-usize-length_bytes-usize---resultusize-decodeerror)
+* [Usage](#usage)
+  + [Serializing/Encoding](#serializingencoding)
+    - [Rust](#rust)
+* [Deserializing/Decoding](#deserializingdecoding)
+    - [Rust](#rust-1)
+
+### TODO
```
Comment (paulhauner) [inline ssz/README.md]: I reckon we can lose these TODOs for now

File: ssz/README.md [hunk 3]
```diff
@@ -1,21 +1,549 @@
...
+
+Convert directly to bytes the size of the int. (e.g. ``int16 = 2 bytes``)
+
+| Check to perform       | Code                    |
+|:-----------------------|:------------------------|
+| Int size is not 0      | ``int_size > 0``        |
+| Size is a byte integer | ``int_size % 8 == 0``   |
+| Value is less than max | ``2**int_size > value`` |
+
+```
```
Comment (paulhauner) [inline ssz/README.md]: Minor point, but if you add "python" after the "```" we'll get syntax highlighting. Not sure if you're intending for this to be actual python or just pseudo-code :)

File: ssz/README.md [hunk 4]
```diff
@@ -1,21 +1,549 @@
...
+assert( len(value) == 32 )
+return value
+```
+
+#### Bytes
+
+For general `byte` type:
+1. Get the length/number of bytes; Encode into a 4byte integer.
```
Comment (paulhauner) [inline ssz/README.md]: "4Byte" might be typo?

File: ssz/README.md [hunk 5]
```diff
@@ -1,21 +1,549 @@
...
+Note: Each return will provide ``deserialized_object, new_index`` keeping track
+of the new index.
+
+At each step, the following checks should be made:
+
+| Check Type               | Check                                                     |
+|:-------------------------|:----------------------------------------------------------|
+| Ensure sufficient length | ``length(rawbytes) > current_index + deserialize_length`` |
+
+#### Int or Uint: 8/16/24/32/64/256
```
Comment (paulhauner) [inline ssz/README.md]: Maybe we should specifically mention that integers should be interpreted as big-endian?

File: ssz/README.md [hunk 6]
```diff
@@ -1,21 +1,549 @@
...
+```
+
+#### List
+
+1. Get the length of the serialized list bytes.
+2. Loop through the bytes;
+   1. Deserialize the object with that length.
+   2. Keep track of current position
+
+Note Before: there are a number of checks to be performed, ensuring there is
```
Comment (paulhauner) [inline ssz/README.md]: I'm not exactly sure what this sentence communicates, sorry

**Comment by NatoliChris** [comment]
@paulhauner 

Updated with comments! Ready for re-review if necessary :)

**Comment by NatoliChris** [comment]
Thanks so much @paulhauner ! 

Great comments! 

Updated to address the comments!

Changes:
* Add syntax highlighting to examples. 
* Fix Typos
* Update ambiguity and old information.

Ready for re-review if necessary :smile:

**Comment by paulhauner** [comment]
Perfect, thanks man!

**Pull request merged**